### PR TITLE
Refs [TASK][38380] Init API

### PR DESCRIPTION
**Pull request by vinhmone**
## Related Tickets
- [#Task 38380: Init API](https://edu-redmine.sun-asterisk.vn/issues/38380)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/findflight/data/source/remote/api/ApiQuery.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    fun queryToken() = Uri.Builder().scheme(ApiConstants.SCHEME_HTTPS)
+        .authority(ApiConstants.AUTHORITY_API_FLIGHT)
+        .appendPath(ApiConstants.CONTENT)
+        .toString()
+
+    fun querySearchNearestAirport(latitude: String, longitude: String) = Uri.Builder()
+        .scheme(ApiConstants.SCHEME_HTTPS)
+        .authority(ApiConstants.AUTHORITY_API_FLIGHT)
+        .appendPath(ApiConstants.CONTENT)
+        .appendEncodedPath(ApiConstants.SEARCH_PLACE)
```
Comment (VietKFC) [inline app/src/main/java/com/sun/findflight/data/source/remote/api/ApiQuery.kt:16]: Nhuững string append giống nhau thế này ông viết ra chung 1 hàm khác để đỡ viết lại nhiều lần nhé

File: app/src/main/java/com/sun/findflight/data/source/remote/api/ApiQuery.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    fun queryToken() = Uri.Builder().scheme(ApiConstants.SCHEME_HTTPS)
+        .authority(ApiConstants.AUTHORITY_API_FLIGHT)
+        .appendPath(ApiConstants.CONTENT)
+        .toString()
+
+    fun querySearchNearestAirport(latitude: String, longitude: String) = Uri.Builder()
+        .scheme(ApiConstants.SCHEME_HTTPS)
+        .authority(ApiConstants.AUTHORITY_API_FLIGHT)
+        .appendPath(ApiConstants.CONTENT)
```
Comment (VietKFC) [inline app/src/main/java/com/sun/findflight/data/source/remote/api/ApiQuery.kt]: những string append mà query nào cũng dùng thì ong viết ra 1 hàm để tái sự dụng nhé

File: app/src/main/java/com/sun/findflight/data/model/Flight.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    val lastTicketingDate: String,
+    val numberOfBookableSeats: Int,
+    val adults: Int = 1,
+    val children: Int = 0,
+    val infants: Int = 0,
+    val currencyCode: String?,
+    val price: Int,
+) : Parcelable {
+
+}
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/data/model/Flight.kt]:
```suggestion
) : Parcelable
```

File: app/src/main/java/com/sun/findflight/data/model/Place.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    val iataCode: String,
+    val cityName: String,
+    val countryName: String,
+    val countryCode: String,
+    val flightScore: Int,
+    val travelerScore: Int,
+    val image: String,
+) : Parcelable {
+
+}
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/findflight/data/model/Place.kt]: tương tự

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**